src/MathOptInterface.jl

```diff
@@ -147,6 +147,27 @@ end
 """
 function copy! end
 
+"""
+    emptycopy(model::ModelLike)
+
+Return an independent empty copy of `model`. This can be used to duplicate a
```

Comment:
What is exactly is copied over, if anything? Solver parameters?

Reply:
What about the license environment?

Reply:
What is the issue with license environment ?
Maybe the easiest is to say that `JuMP` uses this by doing `emptycopy` followed by `copy!` so everything that will not be carried over by `copy!` need to be copied by `emptycopy`.

Reply:
Gurobi has an `Env` that needs to be copied when you create a new model. The `Env` has both license information and solver parameters.

Reply:
Strictly speaking, the solver parameters should be reset to default but the LICENSE should be copied.

Reply:
Is this practical to implement? @odow @joaquimg

Reply:
If this is still useful, should I open an issue? Or can this PR be closed without one.

Reply:
Not sure, we could just close, there is no need for it even if it could be useful